celery/worker/job.py

```diff
@@ -437,9 +437,6 @@ def on_failure(self, exc_info):
                     self._announce_revoked(
                         'terminated', True, string(exc), False)
                     send_failed_event = False  # already sent revoked event
-            # (acks_late) acknowledge after result stored.
-            if self.task.acks_late:
```

Comment:
I would not suggest to remove this, just create a variable inside celery task "celery/app/task.py" | dont_ack_on_failure as boolean value, and put that check here 
if self.task.acks_late  or task.dont_ack_on_failure:
    self.acknowledge()

Reply:
Yes we can add a new variable also. 
Removed this check as we would never want to acknowledge on failure.

Reply:
and we will pass this `dont_ack_on_failure=True` in annotation, right ?

Reply:
I am not sure, what will happen to existing tasks if we remove this check
so adding new variable should be safe

Reply:
@mpsmks existing tasks will not be affected as the existing variable `acks_late` is not used anywhere in espresso.

Reply:
but yes having both `late_ack=True` and `dont_ack_on_failure=True` gives extra functionality. 
@Nitindunzo can u add the new one

Reply:
wrong nitin I guess, @nitinj12388 ?